nifi-nar-bundles/nifi-smb-bundle/nifi-smb-smbj-client/src/main/java/org/apache/nifi/services/smb/SmbjClientProviderService.java

```diff
@@ -0,0 +1,183 @@
+            .displayName("Domain")
+            .name("domain")
+            .description(
+                    "The domain used for authentication. Optional, in most cases username and password is sufficient.")
+            .required(false)
+            .addValidator(NON_EMPTY_VALIDATOR)
+            .build();
+    public static final PropertyDescriptor USERNAME = new PropertyDescriptor.Builder()
+            .displayName("Username")
+            .name("username")
+            .description(
+                    "The username used for authentication.")
+            .required(false)
+            .defaultValue("Guest")
+            .addValidator(NON_EMPTY_VALIDATOR)
+            .build();
+    public static final PropertyDescriptor PASSWORD = new PropertyDescriptor.Builder()
+            .displayName("Password")
+            .name("password")
+            .description("The password used for authentication.")
+            .required(false)
+            .addValidator(NON_EMPTY_VALIDATOR)
+            .sensitive(true)
+            .build();
+    public static final PropertyDescriptor PORT = new PropertyDescriptor.Builder()
+            .displayName("Port")
+            .name("port")
+            .description("Port to use for connection.")
+            .required(true)
+            .addValidator(PORT_VALIDATOR)
+            .defaultValue("445")
+            .build();
+    public static final PropertyDescriptor SHARE = new PropertyDescriptor.Builder()
+            .displayName("Share")
+            .name("share")
+            .description("The network share to which files should be listed from. This is the \"first folder\"" +
+                    "after the hostname: smb://hostname:port\\[share]\\dir1\\dir2")
+            .required(true)
+            .addValidator(NON_BLANK_VALIDATOR)
+            .build();
+    public static final PropertyDescriptor TIMEOUT = new PropertyDescriptor.Builder()
+            .displayName("Timeout")
+            .name("timeout")
+            .description("Timeout for read and write operations.")
+            .required(true)
+            .defaultValue("5 secs")
+            .addValidator(TIME_PERIOD_VALIDATOR)
+            .build();
+    private static final List<PropertyDescriptor> PROPERTIES = Collections
+            .unmodifiableList(asList(
+                    HOSTNAME,
+                    PORT,
+                    SHARE,
+                    USERNAME,
+                    PASSWORD,
+                    DOMAIN,
+                    TIMEOUT
+            ));
+    private SMBClient smbClient;
+    private AuthenticationContext authenticationContext;
+    private String hostname;
+    private int port;
+    private String shareName;
+
+    @Override
+    public SmbClientService getClient() {
+        try {
+            final Connection connection = smbClient.connect(hostname, port);
+            final Session session = connection.authenticate(authenticationContext);
```

Comment:
@kulikg It seems the connection can become stale and unusable under some circumstances. In these cases, `connection.isConnected()` still returns `true` (that's why `SMBClient` does not create a new one) but `connection.authenticate(...)` fails.

I ran into the following cases:

1. Using SMB server running in a Docker container locally. After 60 seconds (even when runnig in every 2 seconds, so no inactivity):

```
com.hierynomus.smbj.common.SMBRuntimeException: com.hierynomus.protocol.transport.TransportException: java.util.concurrent.ExecutionException: com.hierynomus.smbj.common.SMBRuntimeException: java.util.concurrent.TimeoutException: Timeout expired
	at com.hierynomus.smbj.connection.SMBSessionBuilder.establish(SMBSessionBuilder.java:114)
	at com.hierynomus.smbj.connection.Connection.authenticate(Connection.java:202)
	at org.apache.nifi.services.smb.SmbjClientProviderService.getClient(SmbjClientProviderService.java:126)
```

2. Using a Windows share. After some idle / inactive time:

```
com.hierynomus.smbj.common.SMBRuntimeException: com.hierynomus.protocol.transport.TransportException: java.net.SocketException: Broken pipe (Write failed)
	at com.hierynomus.smbj.connection.SMBSessionBuilder.establish(SMBSessionBuilder.java:114)
	at com.hierynomus.smbj.connection.Connection.authenticate(Connection.java:202)
	at org.apache.nifi.services.smb.SmbjClientProviderService.getClient(SmbjClientProviderService.java:126)
```

It seems to be a bug in the smbj library or some extra config may help.
As a workaround, the exceptions above could be caught and the connection could be caused forcefully and then `SMBClient.connect()` would create a new one.